### ISSUE TITLE
Yubii Home support

### ISSUE THREAD

**Issue by coucou2000** [ilcato/homebridge-fibaro-home-center#88]
Hi,

Do you support the Yubii Home which is very close to the FC3 lite.

Thanks

**Comment by coucou2000** [comment]
I tried with the last version, it doesn't work. No accessories loaded in the HomeBridge.

**Comment by ilcato** [comment]
@coucou2000, is the plugin able to connect to your Subii Home ? Unfortunately I do not have such a device.

**Comment by coucou2000** [comment]
@ilcato, yes the plugin is able to connect to my Yubii Home without problem, but no device loaded. Here is exemple log :

[3/1/2022, 5:47:39 PM] [Homebridge UI] Changes to config.json saved.
[3/1/2022, 5:47:39 PM] [Homebridge UI] Running Command: npm uninstall --save homebridge-fibaro-hc3
[3/1/2022, 5:47:44 PM] [Homebridge UI] Homebridge restart request received
[3/1/2022, 5:47:44 PM] [Homebridge UI] UI / Bridge settings have not changed; only restarting Homebridge process
[3/1/2022, 5:47:44 PM] [Homebridge UI] Sending SIGTERM to Homebridge
[3/1/2022, 5:47:44 PM] Got SIGTERM, shutting down Homebridge...
[3/1/2022, 5:47:49 PM] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[3/1/2022, 5:47:54 PM] [HB Supervisor] Restarting Homebridge...
[3/1/2022, 5:47:54 PM] [HB Supervisor] Starting Homebridge with extra flags: -I -P /homebridge/node_modules
[3/1/2022, 5:47:54 PM] [HB Supervisor] Started Homebridge v1.4.0 with PID: 5581
[3/1/2022, 5:47:55 PM] Loaded config.json with 1 accessories and 2 platforms.
[3/1/2022, 5:47:55 PM] Loaded 0 cached accessories from cachedAccessories.
[3/1/2022, 5:47:55 PM] ---
[3/1/2022, 5:47:56 PM] Loaded plugin: homebridge-dummy@0.6.0
[3/1/2022, 5:47:56 PM] Registering accessory 'homebridge-dummy.DummySwitch'
[3/1/2022, 5:47:56 PM] ---
[3/1/2022, 5:47:56 PM] Loaded plugin: homebridge-fibaro-home-center@1.2.8-beta.0
[3/1/2022, 5:47:56 PM] Registering platform 'homebridge-fibaro-home-center.FibaroHC'
[3/1/2022, 5:47:56 PM] ---
[3/1/2022, 5:47:56 PM] Loaded plugin: homebridge-config-ui-x@4.42.0
[3/1/2022, 5:47:56 PM] Registering platform 'homebridge-config-ui-x.config'
[3/1/2022, 5:47:56 PM] ---
[3/1/2022, 5:47:56 PM] Loading 2 platforms...
[3/1/2022, 5:47:56 PM] [Config] Initializing config platform...
[3/1/2022, 5:47:56 PM] [Config] Running in Service Mode
[3/1/2022, 5:47:56 PM] [FibaroHC] Initializing FibaroHC platform...
[3/1/2022, 5:47:57 PM] Loading 1 accessories...
[3/1/2022, 5:47:57 PM] [Test] Initializing DummySwitch accessory...
[3/1/2022, 5:47:57 PM] [Test] Setting switch to false
Setup Payload:
X-HM://0023ISYWYSHME
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 053-76-381            │     
    └────────────┘     
                       
[3/1/2022, 5:47:57 PM] Homebridge v1.4.0 (HAP v0.10.0) (Homebridge E7E3) is running on port 52014.
[3/1/2022, 5:47:57 PM] [FibaroHC] Loading accessories

**Comment by ilcato** [comment]
@coucou2000, are you able to call the rest api from the Yubii interface ?

**Comment by coucou2000** [comment]
@ilcato can you show me how to do it ?

**Comment by ilcato** [comment]
@coucou2000, 
in the lower left-hand corner of the interface there are four buttons: Settings, Console, API documentation, Support.
Select API documentation and access the interactive REST API documentation interface (Swagger).

From there you can call all the API. I'm interesting in what happens by calling the get devices api.

**Comment by coucou2000** [comment]
@ilcato when i click Swagger button, it begins to load the http://192.168.1.110/swagger page. But for a few minutes, it's always in the loading status with the blanc page. Nothing appears in the page.

**Comment by coucou2000** [comment]
@ilcato I finally get the page loaded, i tried the GET ​/devices API, it seems OK with the 200 return code, here is a part of the result :

[
  {
    "id": 1,
    "name": "zwave",
    "roomID": 219,
    "view": [],
    "type": "com.fibaro.zwavePrimaryController",
    "baseType": "",
    "enabled": true,
    "visible": false,
    "isPlugin": false,
    "parentId": 0,
    "viewXml": false,
    "configXml": false,
    "interfaces": [
      "zwave"
    ],
    "properties": {
      "UIMessageSendTime": 0,
      "autoConfig": 0,
      "configured": true,
      "date": "a",
      "dead": false,
      "deviceControlType": 1,
      "deviceIcon": 28,
      "deviceRole": "Other",
      "disabled": 1,
      "emailNotificationID": 0,
      "emailNotificationType": 0,
      "endPoint": 0,
      "endPointId": 0,
      "liliOffCommand": "",
      "liliOnCommand": "",
      "log": "",
      "logTemp": "",
      "manufacturer": "",
      "markAsDead": true,
      "model": "",
      "nodeID": 1,
      "nodeId": 1,
      "parameters": [],
      "parametersTemplate": 0,
      "pollingDeadDevice": false,
      "pollingTime": 1354829138,
      "pollingTimeNext": 1354837724,
      "pollingTimeSec": 125,
      "productInfo": "",
      "pushNotificationID": 0,
      "pushNotificationType": 0,
      "remoteGatewayId": 0,
      "requestNodeNeighborStat": 0,
      "requestNodeNeighborStatTimeStemp": "",
      "requestNodeNeighborState": "",
      "requestNodeNeighborStateTimeStemp": "",
      "saveLogs": true,
      "serialNumber": "",
      "showChildren": 1,
      "smsNotificationID": 0,
      "smsNotificationType": 0,
      "status": "STAT_IDLE",
      "sunriseHour": "06:39",
      "sunsetHour": "17:30",
      "useTemplate": true,
      "userDescription": "",
      "value": 0,
      "zwaveBuildVersion": "3.67",
      "zwaveCompany": "Unknown",
      "zwaveInfo": "",
      "zwaveRegion": "EU",
      "zwaveVersion": "4.33"
    },
    "actions": {
      "pollingDeadDevice": 1,
      "pollingTimeSec": 1,
      "reconfigure": 0,
      "requestNodeNeighborUpdate": 1,
      "turnOff": 0,
      "turnOn": 0
    },
    "created": 1643742562,
    "modified": 1643742562,
    "sortOrder": 1
  },
  {
    "id": 2,
    "name": "admin",
    "roomID": 219,
    "view": [],
    "type": "HC_user",
    "baseType": "com.fibaro.voipUser",
    "enabled": true,
    "visible": true,
    "isPlugin": false,
    "parentId": 0,
    "viewXml": false,
    "configXml": false,
    "interfaces": [
      "voip"
    ],

**Comment by ilcato** [comment]
How many ids are returned?

**Comment by coucou2000** [comment]
@ilcato i  all, 12 ids were returned.

**Comment by coucou2000** [comment]
@ilcato in fact, it's a roller shutter device made by NICE : https://www.niceforyou.com/en/solutions/automation-for-shutters.

**Comment by ilcato** [comment]
@coucou2000, can you test 1.2.8-beta.2 ?

**Comment by coucou2000** [comment]
@ilcato super, it works !

Just a question, i can up/down the shutter but can not have the STOP button. It's normal ?

**Comment by coucou2000** [comment]
i can use Siri to up/down completely the shutter, but when i asked to reach the % level, nothing happened, the commande is just ignored.

**Comment by coucou2000** [comment]
I am French-speaking, for example, once translated, I ask SIRI to "lower the shutter of the living room to 50%"

**Comment by ilcato** [comment]
With the Fibaro roller shutter it works.
Can you post the entire message of the com.fibaro.remoteBaseShutter device ?

**Comment by ilcato** [comment]
No, I mean the one obtained from the API, like:
`{
"id": 1,
"name": "zwave",
"roomID": 219,
"view": [],
"type": "com.fibaro.zwavePrimaryController",
"baseType": "",
"enabled": true,
"visible": false,
"isPlugin": false,
"parentId": 0,
"viewXml": false,
"configXml": false,
"interfaces": [
"zwave"
],`

**Comment by coucou2000** [comment]
{
    "id": 33,
    "name": "Volet",
    "roomID": 221,
    "view": [
      {
        "assetsPath": "/dynamic-plugins/com.fibaro.remoteBaseShutter/assets",
        "jsPath": "/dynamic-plugins/com.fibaro.remoteBaseShutter",
        "name": "com.fibaro.remoteBaseShutter",
        "translatesPath": "/dynamic-plugins/com.fibaro.remoteBaseShutter/i18n",
        "type": "ts"
      }
    ],
    "type": "com.fibaro.remoteBaseShutter",
    "baseType": "com.fibaro.remoteController",
    "enabled": true,
    "visible": true,
    "isPlugin": false,
    "parentId": 7,
    "viewXml": false,
    "configXml": false,
    "interfaces": [
      "nice",
      "niceMono"
    ],
    "properties": {
      "buttonHold": 5000,
      "categories": [
        "blinds",
        "remotes"
      ],
      "configuration": true,
      "dead": false,
      "deadReason": "",
      "deviceControlType": 53,
      "deviceIcon": 218,
      "deviceRole": "BlindsWithoutPositioning",
      "inputToChannelMap": {
        "close": [
          3
        ],
        "open": [
          1
        ],
        "partialOpen1": [],
        "step": [],
        "stop": [
          2
        ],
        "toggleCh1": [],
        "toggleCh2": [],
        "turnOffCh1": [],
        "turnOffCh2": [],
        "turnOnCh1": [],
        "turnOnCh2": [],
        "unsupported": []
      },
      "log": "",
      "logTemp": "",
      "manufacturer": "NICE",
      "model": "",
      "niceId": 2,
      "niceProtocol": "Opera0",
      "numberOfSupportedButtons": 8,
      "saveLogs": true,
      "supportedDeviceRoles": [
        "BlindsWithoutPositioning",
        "VenetianBlinds",
        "Awning"
      ],
      "userDescription": ""
    },
    "actions": {
      "close": 0,
      "open": 0,
      "stop": 0
    },
    "created": 1646228877,
    "modified": 1646228877,
    "sortOrder": 7
  },

**Comment by ilcato** [comment]
Let me check tonight how the Fibaro Roller shutter is different from this perspective.

**Comment by coucou2000** [comment]
Thanks a lot. Alternatively, it would be better if it's possible to implement the 3rd STOP button as done in the HC3 gateway.

![HC3](https://user-images.githubusercontent.com/98122366/156383235-2253237c-774c-4d5c-9735-6418d8cb5ca9.png)

**Comment by ilcato** [comment]
Unfortunately the Home app doesn't expose the stop UI.

**Comment by ilcato** [comment]
@coucou2000, can you try to execute the following command in the swagger interface?
![image](https://user-images.githubusercontent.com/1260538/156428479-e7848987-ea70-49d3-a2a8-0b4282a9539a.png)

**Comment by coucou2000** [comment]
@ilcato the following response : 

202
Undocumented
	
Response body

Download
{
  "id": 39,
  "timestamp": 1646262443
}

Response headers

 cache-control: no-cacheno-store 
 connection: keep-alive 
 content-type: application/json;charset=UTF-8 
 date: Wed02 Mar 2022 23:07:23 GMT 
 server: nginx 
 transfer-encoding: chunked

**Comment by ilcato** [comment]
@coucou2000, did the device changed its position ?

**Comment by coucou2000** [comment]
@ilcato every time i run the command, the shutter pull up to the top of position. If it's already on the top, nothing happens.

**Comment by ilcato** [comment]
@coucou2000, ok. If you put a 0 instead 30 in the args field does the shutter close ?

**Comment by coucou2000** [comment]
@ilcato yes, it closed to the bottom.

**Comment by ilcato** [comment]
@coucou2000, so it seems that the device is not controlled with that parameter for the intermediate values (like the Fibaro one).
The other thing to notice is that it seems to support only BlindsWithoutPositioning role. The Fibaro device supports also BlindsWithPositioning.
Which product is it exactly ? Can you indicate the corresponding web site page ?

**Comment by coucou2000** [comment]
@ilcato that is what i think. 

It's an old mechanical shutter in which I added some parts to electrify it: a motor (ERM08000000), a receiver (TT1N) and a remote control (ERA 1). Everything comes from Nice, a company based in Italy. Perhaps it is not very present in English-speaking countries. Nevertheless, I found some links for related products.

NICE TT1N controler https://www.niceforyou.com/en/videos-and-faq/control-systems/tt1n
NICE ERA  P transmitter https://www.niceforyou.com/en/videos-and-faq/control-systems/era-p-era-w
NICE Motor : https://www.niceforyou.com/en/videos-and-faq/shutters/era-mat-a

![image](https://user-images.githubusercontent.com/98122366/156532257-0741a8b1-0391-440d-93db-087d83cd49d2.png)
